tornado/ioloop.py

```diff
@@ -625,6 +632,35 @@ def add_future(self, future, callback):
         future.add_done_callback(
             lambda future: self.add_callback(callback, future))
 
+    def run_in_executor(self, executor, func, *args):
+        """Runs a function in a ``concurrent.futures.Executor``. If
+        ``executor`` is ``None``, the IO loop's default executor will be used.
+
+        Use `functools.partial` to pass keyword arguments to `func`.
+
+        """
+        if not _concurrent_futures_found:
```

Comment:
Elsewhere for this pattern we just use `if ThreadPoolExecutor is None` instead of a separate boolean variable.